Source/WebCore/Modules/mediasession/MediaSession.cpp

```diff
@@ -268,6 +269,7 @@ void MediaSession::callActionHandler(const MediaSessionActionDetails& actionDeta
 
 bool MediaSession::callActionHandler(const MediaSessionActionDetails& actionDetails, TriggerGestureIndicator triggerGestureIndicator)
 {
+    Locker lock { m_actionHandlersLock };
```

Comment:
We should consider reducing the scope of the lock. Something like:
```
RefPtr<MediaSessionActionHandler> handler;
{
    Locker lock { m_actionHandlersLock };
    handler = m_actionHandlers.get(actionDetails.action);
}
if (handler) {
```

Reply:
Yes! Will do.